pkg/daemon/update.go

```diff
@@ -622,14 +720,85 @@ func (dn *Daemon) update(oldConfig, newConfig *mcfgv1.MachineConfig, skipCertifi
 	if err != nil {
 		return err
 	}
+	err = upgrademonitor.GenerateAndApplyMachineConfigNodes(
+		&upgrademonitor.Condition{State: mcfgalphav1.MachineConfigNodeUpdatePrepared, Reason: string(mcfgalphav1.MachineConfigNodeUpdateCompatible), Message: "Update is Compatible."},
+		&upgrademonitor.Condition{State: mcfgalphav1.MachineConfigNodeUpdateCompatible, Reason: fmt.Sprintf("%s%s", string(mcfgalphav1.MachineConfigNodeUpdatePrepared), string(mcfgalphav1.MachineConfigNodeUpdateCompatible)), Message: fmt.Sprintf("Update Compatible. Post Cfg Actions %v: Drain Required: %t", actions, drain)},
+		metav1.ConditionTrue,
+		metav1.ConditionTrue,
+		dn.node,
+		dn.mcfgClient,
+		dn.featureGatesAccessor,
+	)
+	if err != nil {
+		klog.Errorf("Error making MCN: %w", err)
+	}
 	if drain {
+		//	annos := map[string]string{
+		//	constants.MachineConfigDaemonStateAnnotationKey:  constants.MachineConfigDaemonStateWorking,
+		//	constants.MachineConfigDaemonReasonAnnotationKey: "DrainingNode",
+		//}
+		//	_, err = dn.nodeWriter.SetAnnotations(annos)
+		//	if err != nil {
+		//		return err
+		//	}
+		//dn.EmitUpgradeEvent(dn.stateControllerPod, dn.UpgradeAnnotations(mcfgv1.MachineConfigNodeUpdateExecuted), corev1.EventTypeNormal, "DrainingNode", fmt.Sprintf("Draining Node %s", dn.nodeName()))
+
 		if err := dn.performDrain(); err != nil {
 			return err
 		}
 	} else {
 		klog.Info("Changes do not require drain, skipping.")
+		err := upgrademonitor.GenerateAndApplyMachineConfigNodes(
+			&upgrademonitor.Condition{State: mcfgalphav1.MachineConfigNodeUpdateExecuted, Reason: string(mcfgalphav1.MachineConfigNodeUpdateDrained), Message: "Node Drain Not required for this upgrade."},
+			&upgrademonitor.Condition{State: mcfgalphav1.MachineConfigNodeUpdateDrained, Reason: fmt.Sprintf("%s%s", string(mcfgalphav1.MachineConfigNodeUpdateExecuted), string(mcfgalphav1.MachineConfigNodeUpdateDrained)), Message: "Node Drain Not required for this upgrade."},
+			metav1.ConditionUnknown,
+			metav1.ConditionFalse,
+			dn.node,
+			dn.mcfgClient,
+			dn.featureGatesAccessor,
+		)
+		if err != nil {
+			klog.Errorf("Error making MCN: %w", err)
+		}
 	}
 
+	files := ""
+	for _, f := range newIgnConfig.Storage.Files {
+		files += f.Path + " "
+	}
+
+	osUpdate := "not"
```

Comment:
Maybe we can consider making the message a list instead of individual "need/not need"